core/versioning/txVersionChecker.go

```diff
@@ -23,7 +19,7 @@ func NewTxVersionChecker(minTxVersion uint32) *txVersionChecker {
 
 // IsSignedWithHash will return true if transaction is signed with hash
 func (tvc *txVersionChecker) IsSignedWithHash(tx *transaction.Transaction) bool {
-	if tx.Version > initialVersionOfTransaction {
+	if tx.Version > core.InitialVersionOfTransaction {
```

Comment:
might explain the combination of the fields / options somewhere:
- version=1;options=null => regular signing (over JSON serialization)
- version=1;options={signUsingHash=true;guardianSet=false} => signing over the hash of the transaction, not guardian type
- version=1;options={signUsingHash=false;guardianSet=true} => regular signing over JSON serialization, guardian type
- version=1;options={signUsingHash=true;guardianSet=true} => signing over the hash of the transaction, guardian type

Reply:
thx, added comment